### PR TITLE
B3 CommCare Exchange

### PR DESCRIPTION
**Pull request by biyeun**
screen shot 2016-03-02 at 11 03 43 pm
screen shot 2016-03-02 at 11 03 49 pm
screen shot 2016-03-02 at 11 03 54 pm
screen shot 2016-03-02 at 11 04 04 pm

### CODE REVIEW

File: corehq/apps/appstore/views.py [hunk 1]
```diff
@@ -99,68 +83,155 @@ def deduplicate(hits):
...
+                    # this avoids putting in snapshots in the list where the
+                    # copied_from domain has been deleted.
+                    d_results.append(domain)
+            except CopiedFromDeletedException as e:
+                notify_exception(
+                    self.request,
+                    message=(
+                        "Fetched Exchange Snapshot Error: {}. "
+                        "The problem snapshot id: {}".format(
+                        e.message, res['_source']['_id'])
```
Comment (dimagimon) [inline corehq/apps/appstore/views.py]: E122 continuation line missing indentation or outdented

File: corehq/apps/appstore/views.py [hunk 2]
```diff
@@ -297,9 +368,9 @@ def inc_downloads(d):
                                                 args=[new_domain.name, new_domain.full_applications()[0].get_id]))
         else:
             messages.error(request, _("You must specify a name for the new project"))
-            return project_info(request, snapshot)
+            return HttpResponseRedirect(reverse(ProjectInformationView.urlname, args=[snapshot]))
     else:
-        return HttpResponseRedirect(reverse('project_info', args=[snapshot]))
+        return  HttpResponseRedirect(reverse(ProjectInformationView.urlname, args=[snapshot]))
```
Comment (dimagimon) [inline corehq/apps/appstore/views.py]: E271 multiple spaces after keyword

File: corehq/apps/appstore/views.py [hunk 3]
```diff
@@ -320,47 +391,81 @@ def project_documentation_file(request, snapshot):
...
             'more_pages': more_pages,
             'include_unapproved': include_unapproved,
             'facet_map': facet_map,
-            'query_str': request.META['QUERY_STRING'],
-            'search_url': reverse('deployments'),
-            'search_query': params.get('search', [""])[0]}
-    return render(request, template, vals)
+            'query_str': self.request.META['QUERY_STRING'],
+            'search_url': reverse(self.urlname),
+            'search_query': self.params.get('search', [""])[0]\
```
Comment (dimagimon) [inline corehq/apps/appstore/views.py]: E502 the backslash is redundant between brackets

File: corehq/apps/appstore/views.py [hunk 4]
```diff
@@ -320,47 +391,81 @@ def project_documentation_file(request, snapshot):
...
+
+    @property
+    @memoized
+    def d_results(self):
+        d_results = [Domain.wrap(res['_source']) for res in self.results['hits']['hits']]
+        return d_results
+
+    @property
+    def page_context(self):
+        more_pages = False if len(self.d_results) <= self.page * 10 else True
```
Comment (orangejenny) [inline corehq/apps/appstore/views.py]: Nit: would be nice to make 10 into a projects_per_page constant

**Comment by orangejenny** [comment]
Looks fine to me pending lint & QA.

**Comment by biyeun** [comment]
qa passed cc @benrudolph

**Comment by benrudolph** [comment]
💰 💰

**Pull request merged**